### ISSUE TITLE
React: Uncaught Syntaxerror: unexpected identifier

### ISSUE THREAD

**Issue by yannbf** [storybookjs/storybook#15233]
**Describe the bug**

Storybook initializes correctly and works fine, then you make a change in a component used in the story, and if you refresh Storybook, it gives the error below:

![image](https://user-images.githubusercontent.com/1671563/121820038-cb1d8b00-cc90-11eb-9812-c22cd0b28e65.png)

**To Reproduce**

1 - Access the [repo](https://github.com/yannbf/mealdrop)
2 - Run `yarn storybook`
3 - Access [CategoryDetailPage - Default story](http://localhost:6006/?path=/story/pages-categorydetailpage--default)
4 - Open `Footer.tsx` and make any change there. Let HMR do its thing
5 - Refresh Storybook. You should see the error.

**Additional context**

I was having issues in my project and I went through multiple versions of Storybook to figure out where the problem was originated from. I ended up finding that the issue was introduced in version 6.3.0-alpha.35, and I think it's related to #14991. The issue is happening from any version within alpha.35 and RC.

cc @bebraw

**Comment by bebraw** [comment]
I can confirm the issue. For reference, here's the full error:

```js
t
/* WEBPACK VAR INJECTION */}.call(this, __webpack_require__(/*! ./node_modules/@pmmmwh/react-refresh-webpack-plugin/lib/runtime/RefreshUtils.js */ "./node_modules/@pmmmwh/react-refresh-webpack-plugin/lib/runtime/RefreshUtils.js"), __webpack_require__(/*! ./node_modules/@pmmmwh/react-refresh-webpack-plugin/overlay/index.js */ "./node_modules/@pmmmwh/react-refresh-webpack-plugin/overlay/index.js")))ry {
    // @ts-ignore
    Review.displayName = "Review";
    // @ts-ignore
    Review.__docgenInfo = { "description": "", "displayName": "Review", "props": { "rating": { "defaultValue": null, "description": "", "name": "rating", "required": false, "type": { "name": "number" } } } };
    // @ts-ignore
    if (typeof STORYBOOK_REACT_CLASSES !== "undefined")
        // @ts-ignore
        STORYBOOK_REACT_CLASSES["src/components/Review/Review.tsx#Review"] = { docgenInfo: Review.__docgenInfo, name: "Review", path: "src/components/Review/Review.tsx#Review" };
}
catch (__react_docgen_typescript_loader_error) { }
```

See especially the line

> t
/* WEBPACK VAR INJECTION */}.call(this, __webpack_require__(/*! ./node_modules/@pmmmwh/react-refresh-webpack-plugin/lib/runtime/RefreshUtils.js */ "./node_modules/@pmmmwh/react-refresh-webpack-plugin/lib/runtime/RefreshUtils.js"), __webpack_require__(/*! ./node_modules/@pmmmwh/react-refresh-webpack-plugin/overlay/index.js */ "./node_modules/@pmmmwh/react-refresh-webpack-plugin/overlay/index.js")))ry {

For a reason I don't understand just yet, it's generating code between the "t" and "ry" of a "try" from a try-catch. The next step is understanding **why** and what does the refresh have to do with this.

Furthermore, it seems the project is using specifically webpack 4, not 5.

**Comment by bebraw** [comment]
The next weird thing is that the generated code is like this:

```js
try {
    // @ts-ignore
    Review.displayName = "Review";
    // @ts-ignore
    Review.__docgenInfo = { "description": "", "displayName": "Review", "props": { "rating": { "defaultValue": null, "description": "", "name": "rating", "required": false, "type": { "name": "number" } } } };
    // @ts-ignore
    if (typeof STORYBOOK_REACT_CLASSES !== "undefined")
        // @ts-ignore
        STORYBOOK_REACT_CLASSES["src/components/Review/Review.tsx#Review"] = { docgenInfo: Review.__docgenInfo, name: "Review", path: "src/components/Review/Review.tsx#Review" };
}
catch (__react_docgen_typescript_loader_error) { }


t
/* WEBPACK VAR INJECTION */}.call(this, __webpack_require__(/*! ./node_modules/@pmmmwh/react-refresh-webpack-plugin/lib/runtime/RefreshUtils.js */ "./node_modules/@pmmmwh/react-refresh-webpack-plugin/lib/runtime/RefreshUtils.js"), __webpack_require__(/*! ./node_modules/@pmmmwh/react-refresh-webpack-plugin/overlay/index.js */ "./node_modules/@pmmmwh/react-refresh-webpack-plugin/overlay/index.js")))ry {
    // @ts-ignore
    Review.displayName = "Review";
    // @ts-ignore
    Review.__docgenInfo = { "description": "", "displayName": "Review", "props": { "rating": { "defaultValue": null, "description": "", "name": "rating", "required": false, "type": { "name": "number" } } } };
    // @ts-ignore
    if (typeof STORYBOOK_REACT_CLASSES !== "undefined")
        // @ts-ignore
        STORYBOOK_REACT_CLASSES["src/components/Review/Review.tsx#Review"] = { docgenInfo: Review.__docgenInfo, name: "Review", path: "src/components/Review/Review.tsx#Review" };
}
catch (__react_docgen_typescript_loader_error) { }


try {
    // @ts-ignore
    Review.displayName = "Review";
    // @ts-ignore
    Review.__docgenInfo = { "description": "", "displayName": "Review", "props": { "rating": { "defaultValue": null, "description": "", "name": "rating", "required": false, "type": { "name": "number" } } } };
    // @ts-ignore
    if (typeof STORYBOOK_REACT_CLASSES !== "undefined")
        // @ts-ignore
        STORYBOOK_REACT_CLASSES["src/components/Review/Review.tsx#Review"] = { docgenInfo: Review.__docgenInfo, name: "Review", path: "src/components/Review/Review.tsx#Review" };
}
catch (__react_docgen_typescript_loader_error) { }
```

It's the same code **three** times except for the center one that has been rewritten a bit my the HMR plugin.

I know react-docgen-typescript-plugin is generating the try/catch at `generateDocgenCodeBlock` but looking at https://github.com/hipstersmoothie/react-docgen-typescript-plugin/compare/v0.7.1...v1.0.0, it wasn't changed directly at least.

I'll do some further research on this. Maybe understanding why it generates the same block three times helps. It likely shouldn't do that.

**Comment by bebraw** [comment]
It seems the project is pulling an old version of react-docgen-typescript-plugin. Looking at `yarn.lock`, I noticed the following

```
"@storybook/preset-create-react-app@^3.1.7":
  version "3.1.7"
  resolved "https://registry.yarnpkg.com/@storybook/preset-create-react-app/-/preset-create-react-app-3.1.7.tgz#e48df77768bdcc4863e3918c11965c45e4110ac8"
  integrity sha512-SR+HGSWCrhHA5sszuIHJYdh2tWNi/zu858WB5RM74OBW4ogo8Bv4/7td4p53eWbdm0zBDbnKcrBmwRrAYqwL9Q==
  dependencies:
    "@pmmmwh/react-refresh-webpack-plugin" "^0.4.3"
    "@types/babel__core" "^7.1.7"
    "@types/webpack" "^4.41.13"
    babel-plugin-react-docgen "^4.1.0"
    pnp-webpack-plugin "^1.6.4"
    react-docgen-typescript-plugin "^0.6.2"
    semver "^7.3.2
```

Then if you inspect `node_modules`, you can see it's using `react-docgen-typescript-plugin "^0.6.2"`.

I'm not sure if this is the problem, though, as it's possible it's still resolving to the right one elsewhere in the tree.

@yannbf Can you simplify the issue further by upgrading or changing the CRA bit? It's possible this is irrelevant as well but it feels a bit strange as it's pulling old dependencies to the project that can potentially break things.

**Comment by yannbf** [comment]
Hey @bebraw thank you so much for the investigation!! I added a resolution step:
```
  "resolutions": {
    "react-docgen-typescript-plugin": "1.0.0"
  },
```

So both storybook and storybook cra preset use the latest version. The problem persisted unfortunately!

**Comment by bebraw** [comment]
@yannbf Ok, good to know. If you have time, you could try to fix it to `0.7.1`. If that fixes the issue, then it's something in `1.0.0`. I'll try this as well unless you get to it first.

**Comment by yannbf** [comment]
> @yannbf Ok, good to know. If you have time, you could try to fix it to `0.7.1`. If that fixes the issue, then it's something in `1.0.0`. I'll try this as well unless you get to it first.

Thanks for your suggestion! That does fix the issue.
